fw/sock_clnt.c

```diff
@@ -292,6 +297,32 @@ static const SsHooks tfw_sock_tls_clnt_ss_hooks = {
 	.connection_recv	= tfw_tls_connection_recv,
 };
 
+/*
+ * We call the same TLS hooks before generic HTTP processing
+ * for both the HTTP/1 and HTTP/2.
+ */
+static const SsProto tfw_sock_listen_protos[] = {
+	{ &tfw_sock_http_clnt_ss_hooks,	TFW_FSM_HTTP},
+	{ &tfw_sock_http_clnt_ss_hooks,	Conn_HttpClnt},
+
+	{ &tfw_sock_tls_clnt_ss_hooks,	TFW_FSM_HTTPS},
+	{ &tfw_sock_tls_clnt_ss_hooks,	Conn_HttpsClnt},
+
+	{ &tfw_sock_tls_clnt_ss_hooks,	TFW_FSM_H2},
+	{ &tfw_sock_tls_clnt_ss_hooks,	Conn_H2Clnt},
+};
+
+static const SsProto *
+tfw_sock_clnt_protos(int type)
```

Comment:
```suggestion
tfw_sock_clnt_proto(int type)
```
This function gives a single `SsProto{}` out of the list of them. So the list (array, actually) can and should be `protos`, while this, I think, should be a single `proto`.